ProcessHacker/memprv.c

```diff
@@ -239,6 +239,65 @@ PPH_MEMORY_ITEM PhpSetMemoryRegionType(
     return memoryItem;
 }
 
+void PhpUpdateHeapRegions(
+    _In_ PPH_MEMORY_ITEM_LIST List,
+    _In_ HANDLE ProcessHandle
+    )
+{
+    PRTL_DEBUG_INFORMATION debugBuffer;
+    HANDLE powerRequestHandle = NULL;
+
+    if (WindowsVersion >= WINDOWS_8)
+    {
+        // The RtlQueryProcessDebugInformation can cause deadlock when querying frozen (suspended) immersive process.
```

Comment:
This comment/workaround is based on comment in `PhQueryProcessHeapInformation`.

I was wondering if there is more info about the issue?
Like what versions are affected and if the problem is still present in recent Win10/Win11.

Reply:
> is more info about the issue?
> what versions are affected and if the problem is still present in recent Win10/Win11.

RtlQueryProcessDebugInformation creates a remote thread for the process but when that process was UWP/Immersive and frozen (suspended) the process/thread cannot be resumed and would deadlock.... It was only an issue on earlier versions of Windows 10 and was fixed by Microsoft.

The function was supposed to create an execution required power request and unfreeze the immersive process before creating the remote thread. So even though Microsoft fixed that issue, some users don't update Windows, so we still have to create our own power request before calling the function.